proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,724 @@
+            "has_holdup": self.config.has_holdup,
+            "area_definition": self.config.area_definition,
+            "property_package": self.config.property_package,
+            "property_package_args": self.config.property_package_args,
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme": self.config.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points
+        })
+
+        feed_side = self.feed_side
+        # Add geometry to feed side
+        feed_side.add_geometry()
+        # Add state blocks to feed side
+        feed_side.add_state_blocks(has_phase_equilibrium=False)
+        # Populate feed side
+        feed_side.add_material_balances(balance_type=self.config.material_balance_type,
+                                        has_mass_transfer=True)
+        feed_side.add_momentum_balances(balance_type=self.config.momentum_balance_type,
+                                        has_pressure_change=self.config.has_pressure_change)
+        # Apply transformation to feed side
+        feed_side.apply_transformation()
+        # Add inlet/outlet ports for feed side
+        self.add_inlet_port(name="feed_inlet", block=feed_side)
+        self.add_outlet_port(name="feed_outlet", block=feed_side)
+        # Make indexed stateblock and separate stateblock for permeate-side and permeate outlet, respectively.
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False  # these blocks are not inlets
+        self.permeate_side = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            self.feed_side.length_domain,
+            doc="Material properties of permeate along permeate channel",
+            default=tmp_dict)
+        self.permeate_out = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of mixed permeate exiting the module",
+            default=tmp_dict)
+        # Add port to permeate_out
+        self.add_port(name="permeate_outlet", block=self.permeate_out)
+
+        # ==========================================================================
+        """ Add references to control volume geometry."""
+        add_object_reference(self, 'length', feed_side.length)
+        add_object_reference(self, 'feed_area_cross', feed_side.area)
+
+        # Add reference to pressure drop for feed side only
+        if (self.config.has_pressure_change is True and
+                self.config.momentum_balance_type != MomentumBalanceType.none):
+            add_object_reference(self, 'deltaP', feed_side.deltaP)
+
+        self._make_performance()
+
+    def _make_performance(self):
```

Comment:
We don't have a similar internal function for variable and constraint creation in the 0D RO model, is this something that was implemented in another 1D model or is necessary for 1D models?

Reply:
It was something that I came across in some IDAES models and seems to be a matter of preference. I implemented that approach here for my own readability, but we can revert this back to the way vars and constraints are implemented in 0DRO (or modify 0DRO to match this). However, I'd prefer to get the 1DRO model working fully before making such changes.

Reply:
Opinion is somewhat divided in IDAES. Most models don't do it, but they are fairly simple and were all written by me. Others seem to prefer having these sub-methods, and they do have some advantages for testing (if you unit test each method separately, which no-one does).